File: app/src/main/java/com/wallace/viewpagerandroid/AlternativeViewContainer.java
package com.wallace.viewpagerandroid;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class AlternativeViewContainer extends FrameLayout {
    protected View defaultView;
    protected View alternativeView;

    public AlternativeViewContainer(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        switch (getChildCount()) {
            case 0:
            case 1:
            case 2:
                defaultView = getChildAt(0);
                if(defaultView != null) {
                    defaultView.setVisibility(VISIBLE);
                }
                alternativeView = getChildAt(1);

                if(alternativeView != null) {
                    alternativeView.setVisibility(GONE);
                }
                break;
        }
    }
}

File: app/src/main/java/com/wallace/viewpagerandroid/AlternativeViewContainer.kt
package com.wallace.viewpagerandroid

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout

open class AlternativeViewContainer(context: Context, attrs: AttributeSet, defStyle: Int) : FrameLayout(context, attrs, defStyle) {

    var defaultView: View? = null
        private set
    var alternativeView: View? = null
        private set

    constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0)

    override fun getAccessibilityClassName(): CharSequence = AlternativeViewContainer::class.java.name

    override fun onFinishInflate() {
        super.onFinishInflate()

        when (childCount) {
            0, 1, 2 -> {
                defaultView = getChildAt(0)
                defaultView?.visibility = View.VISIBLE
                alternativeView = getChildAt(1)
                alternativeView?.visibility = View.GONE
            }
            else -> {
                throw IllegalStateException("The ${AlternativeViewContainer::class.simpleName} should only receive the default view and the alternative view")
            }
        }
    }
}